File: dbtool-frontend/interfaces/ChangeResults.ts
interface ChangeRowDetails {
  rows: Record<string, string>[];
}

interface ChangeResults {
  inserted: { [key: string]: ChangeRowDetails };
  updated: { [key: string]: ChangeRowDetails };
  deleted: { [key: string]: ChangeRowDetails };
}

export type { ChangeResults, ChangeRowDetails };

File: dbtool-backend/DbToolApi/Models/ChangeResults.cs
namespace DbToolApi.Models;

public class ChangeRowDetails
{
    public List<Dictionary<string, string>> Rows { get; set; } = [];
}

public class ChangeResults
{
    public Dictionary<string, ChangeRowDetails> Inserted { get; set; } = [];
    public Dictionary<string, ChangeRowDetails> Updated { get; set; } = [];
    public Dictionary<string, ChangeRowDetails> Deleted { get; set; } = [];
}
